reconcile.go

```diff
@@ -140,13 +159,23 @@ func (a *Autopilot) applyDemotions(state *State, changes RaftChanges) (bool, err
 			continue
 		}
 
+		if demoteSingleFailedServer && srv.Health.Healthy {
+			a.logger.Debug("Ignoring demotion of healthy server during failed server demotion process", "id", change)
+			continue
+		}
+
 		a.logger.Info("Demoting server", "id", srv.Server.ID, "address", srv.Server.Address, "name", srv.Server.Name)
 
 		if err := a.demoteVoter(srv.Server.ID); err != nil {
 			return true, fmt.Errorf("failed demoting server %s: %v", srv.Server.ID, err)
 		}
 
 		demoted = true
+
+		// We only want to demote one failed server at a time to prevent violating the minimum quorum setting.
+		if demoteSingleFailedServer {
```

Comment:
What if we failed to demote it above? i.e. `demoted = false` 🤔 . I'm not sure under what circumstances that could happen.

I think the answer is that we should probably still limit our selves to 1 because the most likely way we could fail to demote I can think of is that we timeout or loose connection before we find out if it worked. In that case we can't assume that `err != nil` means that it actually failed so continuing to attempt another unhealthy demotion could violate the "only one" limit.

I guess the result is that we'd return the error stop the reconcile loop and then try again in 10 seconds.

Yeah overall I agree this is the most sensible apporach, just had to think it through!